### PR TITLE
Implemented: Spinner in Timezone Model for Timezone Configuration (#210)

### LINKED ISSUE #210: Implement Spinner in Timezone Model for Timezone Configuration
## Current behavior
In the timezone configuration section, it's crucial to provide visual feedback to users while the application fetches and processes timezone data. Currently, there is a lack of visual cues, leading to potential user confusion. To enhance the user experience, we propose implementing a spinner in the timezone model. This spinner will indicate loading activity, ensuring users are aware of the system processing their request.

## Expected behavior
The spinner is visible within the timezone model when the application is fetching timezone data and disappears once the timezone data has been successfully loaded and displayed in the timezone configuration section.

## Can you handle fixing this bug by yourself?

- [ ] YES
- [ ] NO

## Environment details

- Browser: 
- OS: 
- Code Version:

### PR DESCRIPTION
**Pull request by Ritika-Patel08**
### Related Issues

Closes #210

### Short Description and Why It's Useful

Spinner is added to show that the timezones are being fetched from the API. It is indicating to the user that the data is loading.

### Contribution and Currently Import
Screenshot 2024-01-06 093850
ant Rules Acceptance

- [x] I have read and followed [contribution rules](https://github.com/hotwax/inventory-count#contribution-guideline)

### CODE REVIEW

File: src/views/TimezoneModal.vue [hunk 1]
```diff
@@ -55,17 +71,19 @@ import {
...
-import { close, save } from "ionicons/icons";
-import { useStore } from "@/store";
-import { UserService } from "@/services/UserService";
-import { hasError } from '@/utils'
-import { DateTime } from 'luxon';
-
-export default defineComponent({
+  alertController
+ } from "@ionic/vue";
+ import { defineComponent } from "vue";
```
Comment (ymaheshwari1) [inline src/views/TimezoneModal.vue:76]: Remove all the irrelevant changes from the PR those are not related to the issue, like adding extra lines, changing single-quotes to double quotes and vice-versa, changing indentation, adding space after keywords.

File: src/views/TimezoneModal.vue [hunk 2]
```diff
@@ -2,44 +2,60 @@
...
+    </div>
+ 
     <!-- Timezones -->
     <div v-else>
       <ion-list>
         <ion-radio-group value="rd" v-model="timeZoneId">
           <ion-item :key="timeZone.id" v-for="timeZone in filteredTimeZones">
-            <ion-radio justify="start" label-placement="end" :value="timeZone.id">{{ timeZone.label }} ({{ timeZone.id }})</ion-radio>
+            <ion-radio justify="start" label-placement="end" :value="timeZone.id">{{ timeZone.label }} ({{ timeZone.id
+            }})</ion-radio>
```
Comment (ymaheshwari1) [inline src/views/TimezoneModal.vue:41]: Unwanted change, please revert.

File: src/views/TimezoneModal.vue [hunk 3]
```diff
@@ -2,44 +2,60 @@
...
+        <ion-item lines="none" color="none">
+          
+          <ion-spinner class="center-spinner" color="secondary" name="crescent" slot="start" ></ion-spinner>
+          {{ $t("Fetching time zones") }}
+        </ion-item>
+      </div>
     </div>
-
+    <div v-if="!loading && filteredTimeZones.length === 0">
+      <div class="empty-state">
```
Comment (ymaheshwari1) [inline src/views/TimezoneModal.vue:30]: No need to have extra div for adding if condition, adding a class and if condition can be handled in a single div.

File: src/views/TimezoneModal.vue [hunk 4]
```diff
@@ -2,44 +2,60 @@
...
+ 
   <ion-content class="ion-padding">
     <!-- Empty state -->
-    <div class="empty-state" v-if="filteredTimeZones.length === 0">
-      <p>{{ $t("No time zone found")}}</p>
+    <div>
+      <div class="empty-state" v-if="loading && timeZones.length === 0">
+        <ion-item lines="none" color="none">
+          
+          <ion-spinner class="center-spinner" color="secondary" name="crescent" slot="start" ></ion-spinner>
```
Comment (ymaheshwari1) [inline src/views/TimezoneModal.vue:24]: Class `center-spinner` is unused.

File: src/views/TimezoneModal.vue [hunk 5]
```diff
@@ -114,23 +133,26 @@ export default defineComponent({
...
+    findTimeZone() {
       const queryString = this.queryString.toLowerCase();
       this.filteredTimeZones = this.timeZones.filter((timeZone: any) => {
         return timeZone.id.toLowerCase().match(queryString) || timeZone.label.toLowerCase().match(queryString);
       });
     },
     async getAvailableTimeZones() {
       const resp = await UserService.getAvailableTimeZones()
-      if(resp.status === 200 && !hasError(resp)) {
+      console.log(resp);
```
Comment (ymaheshwari1) [inline src/views/TimezoneModal.vue:146]: Remove console statement from the code, as we use this console statements for debugging hence should not be pushed.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/locales/en.json [hunk 6]
```diff
@@ -37,6 +37,7 @@
   "Make sure you've reviewed the products and their counts before uploading them for review": "Make sure you've reviewed the products and their counts before uploading them for review",
   "No results found": "No results found",
   "No time zone found": "No time zone found",
+  "Fetching time zones":"Fetching time zones",
```
Comment (amansinghbais) [inline src/locales/en.json:40]:
```suggestion
  "Fetching time zones": "Fetching time zones",
```

File: src/views/TimezoneModal.vue [hunk 7]
```diff
@@ -15,8 +15,14 @@
 
   <ion-content class="ion-padding">
     <!-- Empty state -->
-    <div class="empty-state" v-if="filteredTimeZones.length === 0">
-      <p>{{ $t("No time zone found")}}</p>
+    <div class="empty-state" v-if="loading">
+      <ion-item lines="none" >
```
Comment (amansinghbais) [inline src/views/TimezoneModal.vue:19]:
```suggestion
      <ion-item lines="none">
```

File: src/views/TimezoneModal.vue [hunk 8]
```diff
@@ -15,8 +15,14 @@
...
+        <ion-spinner color="secondary" name="crescent" slot="start" ></ion-spinner>
```
Comment (amansinghbais) [inline src/views/TimezoneModal.vue:20]: Use self closing tag instead.

File: src/views/TimezoneModal.vue [hunk 9]
```diff
@@ -131,6 +138,7 @@ export default defineComponent({
         });
         this.findTimeZone();
       }
+    this.isLoading = false;
```
Comment (ymaheshwari1) [inline src/views/TimezoneModal.vue:141]: Improve identation.

**Review by ymaheshwari1 — APPROVED** [review]

**Pull request merged**